frontend/src/components/map/map.jsx

```diff
@@ -134,48 +137,50 @@ class Map extends React.Component {
       }).setHTML(
         `${type} delivery${`<br />`}${typeIconString(type.toLowerCase(), status)}`
       )
-      // make a marker for each feature and add to the map
+
+      // Make a marker for each feature and add to the map
       const mapBoxMarker = new mapboxgl.Marker(el)
         .setLngLat(marker.geometry.coordinates)
         .setPopup(popup)
+
       // Add mapBox marker and associated id to array
       allMarkers.push({ mBMarker: mapBoxMarker, id: marker.properties.taskId });
 
       const { receiveActiveTaskId } = this.props;
       const markerEl = mapBoxMarker.getElement();
       markerEl.addEventListener('mouseenter', () => {
+
         // Add popup to map 
         popup.addTo(map);
       });
       markerEl.addEventListener('mouseleave', () => {
+
         // Remove popup from map
-        // receiveActiveTaskId(null);
         const { activeTask } = this.props;
-        if (!activeTask || activeTask.taskId !== taskId) {
+        if (popup.isOpen() && (!activeTask || activeTask.taskId !== taskId)) {
```

Comment:
Yea good job on this. Was thinking the same thing when i looked at it last night